mempool/config.go

```diff
@@ -0,0 +1,159 @@
+package mempool
+
+import (
+	"cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+type (
+	// isAuctionTx defines a function that returns true iff a transaction is an
+	// auction bid transaction.
+	IsAuctionTx func(tx sdk.Tx) (bool, error)
+
+	// getTransactionSigners defines a function that returns the signers of a
+	// transaction.
+	GetTransactionSigners func(tx []byte) (map[string]bool, error)
+
+	// wrapBundleTransaction defines a function that wraps a transaction that is included
+	// in the bundle into a sdk.Tx.
+	WrapBundleTransaction func(tx []byte) (sdk.Tx, error)
+
+	// GetBidder defines a function that returns the bidder of a transaction.
+	GetBidder func(tx sdk.Tx) (sdk.AccAddress, error)
+
+	// GetBid defines a function that returns the bid of a transaction.
+	GetBid func(tx sdk.Tx) (sdk.Coin, error)
+
+	// GetBundledTransactions defines a function that returns the bundled transactions
+	// that the user wants to execute at the top of the block.
+	GetBundledTransactions func(tx sdk.Tx) ([][]byte, error)
+
+	// BidInfo defines the information about a bid.
+	BidInfo struct {
+		Bidder       sdk.AccAddress
+		Bid          sdk.Coin
```

Comment:
I would define `TransactionConfig` as an interface and move all the methods into that interface.

nit: i'd probably rename it to `Config` too

Reply:
asking to learn: Is there a reason we want to make `Config` an interface instead of a struct?

Reply:
I think the dev UX is cleaner. Instead of having to pass a concrete struct around you can allow the user to define a single type. In addition, our default config's constructor is more concise.